packaging/suse/portusctl/lib/helper.rb

```diff
@@ -0,0 +1,40 @@
+def check_setup_flags(options)
+  if options["ssl-gen-self-signed-certs"] && \
+      !options["ssl-certs-dir"].chomp.empty?
+    warn "cannot use both options --ssl-gen-self-signed-certs and " \
+      "--ssl-certs-dir at the same time"
+    exit 1
+  end
+
+  return unless options["ldap-enable"] && \
+      (options["ldap-hostname"].nil? || options["ldap-hostname"].empty?)
+
+  warn "LDAP support is enabled but you didn't specify a value for ldap-hostname"
+  exit 1
+end
+
+def ensure_root
+  return if Process.uid == 0
+
+  warn "Must run as root user"
+  exit 1
+end
+
+def handle_own_certs(path)
+  puts "Using keys from #{path}"
+  key_file = File.join(path, "#{HOSTNAME}-ca.key")
+  crt_file = File.join(path, "#{HOSTNAME}-ca.crt")
+
+  missing_file(key_file) unless File.exist?(key_file)
+  FileUtils.cp(key_file, "/etc/apache2/ssl.key")
```

Comment:
I'm not entirely sure about the usage of this method shared between the regular certificate code and the "bring your own certs".

The output in this case would be a bit confusing, like "Use --ssl-gen-self-signed-certs <...> or --ssl-certs-dir".

I would prefer something more clear like: "Cannot find hostname-ca.crt inside of custom_dir".